### PR TITLE
Introduce new metrics through the logs

### PR DESCRIPTION
**Pull request by paraita**
It is now possible to track the state of the jobs in memory to better diagnose performance issues and possible leaks.

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 78 issues
* MAJOR 3 major
* MINOR 75 minor

#### Top 10 issues

1. MAJOR [SchedulerStateImpl.java#L99](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L99): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1149)
1. MAJOR [SchedulerStateImpl.java#L117](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L117): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1149)
1. MAJOR [SchedulerStateImpl.java#L135](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L135): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1149)
1. MINOR [SchedulerFrontend.java#L389](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L389): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L417](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L417): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L427](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L427): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L438](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L438): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L473](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L473): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L483](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L483): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L493](https://github.com/ow2-proactive/scheduling/blob/165f48c421f6df52cb6eb602f9163414f1ce0bb2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L493): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by paraita** [comment]
retest this please

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 78 issues
* MAJOR 3 major
* MINOR 75 minor

#### Top 10 issues

1. MAJOR [SchedulerStateImpl.java#L99](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L99): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1149)
1. MAJOR [SchedulerStateImpl.java#L117](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L117): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1149)
1. MAJOR [SchedulerStateImpl.java#L135](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L135): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1149)
1. MINOR [SchedulerFrontend.java#L389](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L389): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L417](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L417): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L427](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L427): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L438](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L438): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L473](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L473): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L483](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L483): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. MINOR [SchedulerFrontend.java#L493](https://github.com/ow2-proactive/scheduling/blob/0077a659f6cda37c0a7386e69823d761b6c44a08/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L493): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/core/properties/PASchedulerProperties.java [hunk 1]
```diff
@@ -271,7 +271,7 @@ negative number indicates that session is infinite (value specified in second) *
             "[%X{job.id}t%X{task.id}@%X{host};%d{HH:mm:ss}] %m %n"),
 
     /** Defines the cron expression for the db sizes polling */
-    SCHEDULER_DB_SIZE_MONITORING_FREQ("pa.scheduler.db.size.monitoring.freq", PropertyType.STRING),
+    SCHEDULER_MEM_MONITORING_FREQ("pa.scheduler.mem.monitoring.freq", PropertyType.STRING),
```
Comment (marcocast) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/core/properties/PASchedulerProperties.java]: do we need to change this property as well in other existing clients?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java [hunk 2]
```diff
@@ -325,6 +329,19 @@ public void initActivity(Body body) {
...
+                metricsMonitorScheduler.schedule(cronExpr,
+                                                 new TableSizeMonitorRunner(dbManager.getTransactionHelper()));
+                metricsMonitorScheduler.schedule(cronExpr,
+                                                 new JobsMemoryMonitorRunner(dbManager.getSessionFactory()
+                                                                                      .getStatistics(),
+                                                                             frontendState.getState()));
+                metricsMonitorScheduler.start();
+            }
+
             // run !!
```
Comment (marcocast) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java]: can you remove this comment?

Comment (paraita) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java]: done !

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java [hunk 3]
```diff
@@ -344,6 +344,10 @@ synchronized SchedulerState getState() throws NotConnectedException, PermissionE
         return getState(false);
     }
 
+    synchronized SchedulerState getStateInternally() {
+        return sState;
+    }
+
     synchronized SchedulerState getState(boolean myJobsOnly) throws NotConnectedException, PermissionException {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 4]
```diff
@@ -83,6 +83,7 @@
 import org.ow2.proactive.scheduler.common.usage.JobUsage;
 import org.ow2.proactive.scheduler.core.account.SchedulerAccount;
 import org.ow2.proactive.scheduler.core.db.TaskData.DBTaskId;
+import org.ow2.proactive.scheduler.core.helpers.JobsMemoryMonitorRunner;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java]: MINOR Remove this unused import 'org.ow2.proactive.scheduler.core.helpers.JobsMemoryMonitorRunner'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 5]
```diff
@@ -83,6 +83,7 @@
...
 import org.ow2.proactive.scheduler.core.helpers.TableSizeMonitorRunner;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java]: MINOR Remove this unused import 'org.ow2.proactive.scheduler.core.helpers.TableSizeMonitorRunner'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java [hunk 6]
```diff
@@ -201,6 +203,8 @@
 
     private SchedulerPortalConfiguration schedulerPortalConfiguration = SchedulerPortalConfiguration.getConfiguration();
 
+    private it.sauronsoftware.cron4j.Scheduler metricsMonitorScheduler;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java]: MINOR Remove the "metricsMonitorScheduler" field and declare it as a local variable in the relevant methods. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java [hunk 7]
```diff
@@ -35,6 +35,7 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java]: MINOR Remove this unused import 'java.util.Set'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java [hunk 8]
```diff
@@ -35,6 +35,7 @@
...
 import java.util.Vector;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java]: MINOR Remove this unused import 'java.util.Vector'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java [hunk 9]
```diff
@@ -66,6 +68,7 @@
 import org.ow2.proactive.scheduler.job.InternalJob;
 import org.ow2.proactive.scheduler.job.JobInfoImpl;
 import org.ow2.proactive.scheduler.policy.Policy;
+import org.ow2.proactive.scheduler.rest.data.JobStateImpl;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java]: MINOR Remove this unused import 'org.ow2.proactive.scheduler.rest.data.JobStateImpl'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java [hunk 10]
```diff
@@ -55,6 +56,7 @@
 import org.ow2.proactive.scheduler.common.job.JobId;
 import org.ow2.proactive.scheduler.common.job.JobInfo;
 import org.ow2.proactive.scheduler.common.job.JobPriority;
+import org.ow2.proactive.scheduler.common.job.JobState;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java]: MINOR Remove this unused import 'org.ow2.proactive.scheduler.common.job.JobState'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/JobsMemoryMonitorRunner.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+    private static Statistics stats;
+
+    private static SchedulerState schedulerState;
+
+    private static final Logger logger = Logger.getLogger(JobsMemoryMonitorRunner.class);
+
+    public JobsMemoryMonitorRunner(Statistics statistics, SchedulerState schedulerState) {
+        statistics.setStatisticsEnabled(true);
+        this.stats = statistics;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/JobsMemoryMonitorRunner.java]: MAJOR Remove this assignment of "stats". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3010)
MAJOR Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/JobsMemoryMonitorRunner.java [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+        this.schedulerState = schedulerState;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/JobsMemoryMonitorRunner.java]: MAJOR Remove this assignment of "schedulerState". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3010)
MAJOR Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 108 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 99 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SchedulerStateImpl.java#L103](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L103): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)
1. ![MAJOR][MAJOR] [SchedulerStateImpl.java#L121](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L121): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)
1. ![MAJOR][MAJOR] [SchedulerStateImpl.java#L139](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L139): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)
1. ![MAJOR][MAJOR] [SchedulerDBManager.java#L1735](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1735): Remove this unused private "taskId" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [SchedulerDBManager.java#L1746](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1746): Remove this unused private "jobId" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L403](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L403): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L435](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L435): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L445](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L445): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L456](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L456): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L491](https://github.com/ow2-proactive/scheduling/blob/8b00e551261f597f1ed8ad4489b9c2e6703cbf9a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L491): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java [hunk 13]
```diff
@@ -336,14 +336,17 @@ public void run() {
     synchronized SchedulerStatus getStatus() throws NotConnectedException, PermissionException {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java [hunk 14]
```diff
@@ -336,14 +336,17 @@ public void run() {
...
         // checking permissions
         checkPermission("getStatus", YOU_DO_NOT_HAVE_PERMISSION_TO_GET_THE_STATUS);
-
         return sState.getStatus();
     }
 
     synchronized SchedulerState getState() throws NotConnectedException, PermissionException {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 104 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 99 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SchedulerStateImpl.java#L103](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L103): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)
1. ![MAJOR][MAJOR] [SchedulerStateImpl.java#L121](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L121): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)
1. ![MAJOR][MAJOR] [SchedulerStateImpl.java#L139](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java#L139): Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)
1. ![MAJOR][MAJOR] [SchedulerDBManager.java#L1735](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1735): Remove this unused private "taskId" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [SchedulerDBManager.java#L1746](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1746): Remove this unused private "jobId" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L394](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L394): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L426](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L426): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L436](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L436): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L447](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L447): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L482](https://github.com/ow2-proactive/scheduling/blob/6800ba38b664afe17931d35d15a635f4732097ae/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L482): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**